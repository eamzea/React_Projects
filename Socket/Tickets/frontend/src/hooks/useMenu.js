import React, { useContext, useEffect } from 'react';
import { UIcontext } from '../context/UIcontext';

const useMenu = hide => {
  const { showMenu, hideMenu } = useContext(UIcontext);

  useEffect(() => {
    hide ? hideMenu() : showMenu();
  }, []);
};

export default useMenu;
