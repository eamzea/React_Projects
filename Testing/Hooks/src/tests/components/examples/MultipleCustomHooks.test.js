import "@testing-library/jest-dom";
import React from "react";
import { renderHook } from "@testing-library/react-hooks";
import { shallow } from "enzyme";
import MultipleCustomHooks from "../../../components/examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Pruebas en <MultipleCustomHooks/>", () => {
  useCounter.mockReturnValue({
    counter: 10,
    increment: () => {},
  });

  test("Debe de mostrarse correctamente", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar la informaciÃ³n", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "Edgar",
          quote: "Hola mundo",
        },
      ],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find("alert").exists()).toBe(false);
    expect(wrapper.find(".mb-0").text().trim()).toBe("Hola mundo");
    expect(wrapper.find("footer").text().trim()).toBe("Edgar");
  });
});
