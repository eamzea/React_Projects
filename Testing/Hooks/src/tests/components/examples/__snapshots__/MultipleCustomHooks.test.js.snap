// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks/> Debe de mostrarse correctamente 1`] = `
<div>
  <p
    className="h1"
  >
    Breaking Bad Quotes
  </p>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    Siguiente quote
  </Button>
</div>
`;
